test/e2e_node/image_volume.go

```diff
@@ -44,33 +44,49 @@ var _ = SIGDescribe("ImageVolume", nodefeature.ImageVolume, func() {
 	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
 
 	const (
-		podName          = "test-pod"
-		containerName    = "test-container"
-		validImageRef    = "quay.io/crio/artifact:v1"
-		invalidImageRef  = "localhost/invalid"
-		volumeName       = "volume"
-		volumePathPrefix = "/volume"
+		podName             = "test-pod"
+		containerName       = "test-container"
+		validImageRef       = "quay.io/crio/artifact:v1"
+		invalidImageRef     = "localhost/invalid"
+		volumeName          = "volume"
+		volumePathPrefix    = "/volume"
+		defaultSELinuxUser  = "system_u"
+		defaultSELinuxRole  = "system_r"
+		defaultSELinuxType  = "svirt_lxc_net_t"
+		defaultSELinuxLevel = "s0:c1,c5"
 	)
 
 	ginkgo.BeforeEach(func(ctx context.Context) {
 		e2eskipper.SkipUnlessFeatureGateEnabled(features.ImageVolume)
 	})
 
 	createPod := func(ctx context.Context, volumes []v1.Volume, volumeMounts []v1.VolumeMount) {
+		var selinuxOptions *v1.SELinuxOptions
+		if selinux.GetEnabled() {
+			selinuxOptions = &v1.SELinuxOptions{
+				User:  defaultSELinuxUser,
+				Role:  defaultSELinuxRole,
+				Type:  defaultSELinuxType,
+				Level: defaultSELinuxLevel,
+			}
+			ginkgo.By(fmt.Sprintf("Using SELinux on pod: %v", selinuxOptions))
+		}
 		pod := &v1.Pod{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      podName,
 				Namespace: f.Namespace.Name,
 			},
 			Spec: v1.PodSpec{
 				RestartPolicy: v1.RestartPolicyAlways,
+				SecurityContext: &v1.PodSecurityContext{
+					SELinuxOptions: selinuxOptions,
+				},
```

Comment:
Would it be OK if these options were set to some default values? Or would this also need a `if selinux.GetEnabled()` check?

Reply:
I'd say we can assume that runtimes do not use the fields at all when SELinux is disabled, therefore we could keep it as is. Do you have thoughts on that?

Reply:
No, that sounds good! Thank you!